spark/v3.5/spark-runtime/NOTICE

```diff
@@ -500,9 +363,52 @@ file:
 This binary artifact includes Project Nessie with the following in its NOTICE
 file:
 
+| NOTICE applicable to Nessie source
```

Comment:
Update Nessie `NOTICE` from the latest version.

Reply:
As in #12145, I think we need Nessie to update license documentation before we can conclude that this is accurate. The current NOTICE for binary artifacts (that we use) still includes a line about GPL.

Reply:
I've updated Nessie `NOTICE` corresponding to 0.102.5 release.